File: WallAI.Core/Worlds/Entities/IReadOnlyWorld2DEntity.cs
using WallAI.Core.Entities;
using WallAI.Core.Interfaces;
using WallAI.Core.Math.Geometry;

namespace WallAI.Core.Worlds.Entities
{
    public interface IReadOnlyWorld2DEntity : IReadOnlyEntity, IHasReadOnlyLocation
    {
        new IReadOnlyWorld2DEntity WithLocationOffset(Point2D offset);
    }
}

File: WallAI.HostedSimulation.Web/app/RemoteTypes/Worlds/Entities/IReadOnlyWorld2DEntity.ts
import { IReadOnlyWorld2D } from "../IReadOnlyWorld2D";
import { IReadOnlyEntity2D } from "RemoteTypes/Entities/IReadOnlyEntity2D";
import { IHasReadOnlyLocation } from "RemoteTypes/Interfaces/IHasReadOnlyLocation";

export interface IReadOnlyWorld2DEntity extends IReadOnlyEntity2D, IHasReadOnlyLocation {
    
}